### PR TITLE
Fluent drawer

### PR DESCRIPTION
**Pull request by JoomFX**
related to: https://github.com/telerik/kendo-themes-private/issues/3

### CODE REVIEW

File: packages/fluent/scss/drawer/_layout.scss [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        border-style: solid;
+        box-sizing: border-box;
+        outline: 0;
+        font-family: var( --kendo-drawer-font-family, #{$kendo-drawer-font-family} );
+        font-size: var( --kendo-drawer-font-size, #{$kendo-drawer-font-size} );
+        line-height: var( --kendo-drawer-line-height, #{$kendo-drawer-line-height} );
+        display: flex;
+        flex-direction: column;
+        flex: 0 0 auto;
+        transition: all 300ms ease-in-out; // sass-lint:disable-line no-transition-all
```
Comment (Juveniel) [inline packages/fluent/scss/drawer/_layout.scss:28]: you can use --kendo-transition here ?

File: packages/fluent/scss/drawer/_theme.scss [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            }
+        }
+
+        &:focus,
+        &.k-focus {
+            @include fill(
+                var( --kendo-drawer-item-focus-text, #{$kendo-drawer-item-focus-text} ),
+                var( --kendo-drawer-item-focus-bg, #{$kendo-drawer-item-focus-bg} )
+            );
+            box-shadow: $kendo-drawer-item-focus-shadow;
```
Comment (Juveniel) [inline packages/fluent/scss/drawer/_theme.scss:56]: use css var

File: packages/fluent/scss/drawer/_theme.scss [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+            .k-icon {
+                color: $kendo-drawer-item-focus-icon-text;
```
Comment (Juveniel) [inline packages/fluent/scss/drawer/_theme.scss:59]: use css var

File: packages/fluent/scss/drawer/_theme.scss [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        &.k-selected {
+            @include fill(
+                var( --kendo-drawer-item-selected-text, #{$kendo-drawer-item-selected-text} ),
+                var( --kendo-drawer-item-selected-bg, #{$kendo-drawer-item-selected-bg} )
+            );
+
+            .k-icon {
+                color: $kendo-drawer-item-selected-icon-text;
```
Comment (Juveniel) [inline packages/fluent/scss/drawer/_theme.scss:78]: use css var

File: packages/fluent/scss/drawer/_variables.scss [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+/// Scrollbar width of the drawer.
+/// @group drawer
+$kendo-drawer-scrollbar-width: 7px !default;
+/// Scrollbar color of the drawer.
+/// @group drawer
+$kendo-drawer-scrollbar-color: rgba(156, 156, 156, .7) !default;
+/// Scrollbar background of the drawer.
+/// @group drawer
+$kendo-drawer-scrollbar-bg: #dedede !default;
```
Comment (Juveniel) [inline packages/fluent/scss/drawer/_variables.scss:47]: use value from the color palette

File: packages/fluent/scss/drawer/_variables.scss [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+/// Count of the drawer item levels.
+/// @group drawer
+$kendo-drawer-item-level-count: 5 !default;
+
+/// Scrollbar width of the drawer.
+/// @group drawer
+$kendo-drawer-scrollbar-width: 7px !default;
+/// Scrollbar color of the drawer.
+/// @group drawer
+$kendo-drawer-scrollbar-color: rgba(156, 156, 156, .7) !default;
```
Comment (Juveniel) [inline packages/fluent/scss/drawer/_variables.scss:44]: use value from the color palette

File: packages/fluent/scss/drawer/_variables.scss [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+/// Scrollbar background of the drawer.
+/// @group drawer
+$kendo-drawer-scrollbar-bg: #dedede !default;
+/// Scrollbar border radius of the drawer.
+/// @group drawer
+$kendo-drawer-scrollbar-radius: 20px !default;
+/// Scrollbar hover color of the drawer.
+/// @group drawer
+$kendo-drawer-scrollbar-hover-color: rgba(156, 156, 156, 1) !default;
```
Comment (Juveniel) [inline packages/fluent/scss/drawer/_variables.scss:53]: use value from the color palette

File: packages/fluent/scss/drawer/_variables.scss [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+/// Line height of the drawer item.
+/// @group drawer
+$kendo-drawer-item-line-height: var( --kendo-line-height-lg, normal ) !default;
+/// Icon size of the drawer item.
+/// @group drawer
+$kendo-drawer-icon-size: var( --kendo-icon-size, 1rem ) !default;
+
+/// Initial width of the mini drawer.
+/// @group drawer
+$kendo-drawer-mini-initial-width: calc(4 * #{$kendo-drawer-item-padding-x} + #{$kendo-drawer-icon-size}) !default;
```
Comment (Juveniel) [inline packages/fluent/scss/drawer/_variables.scss:73]: spacing

File: packages/fluent/scss/drawer/index.scss [hunk 9]
```diff
@@ -0,0 +1,26 @@
+// Module meta
+$_kendo-module-meta: (
+    name: "drawer",
+    dependencies: (
+        "overlay",
+        "icon"
```
Comment (Juveniel) [inline packages/fluent/scss/drawer/index.scss:6]: please check if we need to add the toolbar and list as deps

**Review by Juveniel — CHANGES_REQUESTED** [review]

File: packages/fluent/scss/drawer/index.scss (hunk 9 above)
Comment (JoomFX) [reply packages/fluent/scss/drawer/index.scss:6]: No other dependencies need to be added.

File: packages/fluent/scss/drawer/_layout.scss (hunk 1 above)
Comment (JoomFX) [reply packages/fluent/scss/drawer/_layout.scss:28]: When I changed it to `--kendo-transition` the opening animation of Drawer disappeared, at least in the visual tests (when refreshing the page).
Therefore I'm leaving it as-is.

**Comment by Juveniel** [comment]
@JoomFX I noticed that the border in the drawer-rtl test is on the opposite direction. Could you take a look ?

**Review by Juveniel — APPROVED** [review]

**Pull request merged**